java/core/src/main/java/co/worklytics/psoxy/SanitizerFactory.java

```diff
@@ -0,0 +1,27 @@
+package co.worklytics.psoxy;
+
+import co.worklytics.psoxy.gateway.ConfigService;
+import co.worklytics.psoxy.gateway.ProxyConfigProperty;
+import co.worklytics.psoxy.impl.SanitizerImpl;
+
+import dagger.assisted.AssistedFactory;
+
+@AssistedFactory
+public interface SanitizerFactory {
+
+    String DEFAULT_SALT = "salt";
+
+    SanitizerImpl create(Sanitizer.Options options);
+
+    //q: right place? mapping config+rules --> Sanitizer.Options isn't implementation-specific
+    default Sanitizer.Options buildOptions(ConfigService config, Rules rules) {
+        return Sanitizer.Options.builder()
+            .rules(rules)
+                    .pseudonymizationSalt(config.getConfigPropertyAsOptional(ProxyConfigProperty.PSOXY_SALT)
+                        .orElse(DEFAULT_SALT))
```

Comment:
I'd say this is critical enough to not have a DEFAULT_SALT... it should "explode" here if no PSOXY_SALT